pkg/ccl/kvccl/kvfollowerreadsccl/followerreads.go

```diff
@@ -203,6 +205,49 @@ func (o *followerReadOracle) useClosestOracle(
 // whether a query can be used with a follower read.
 var followerReadOraclePolicy = replicaoracle.RegisterPolicy(newFollowerReadOracle)
 
+type bulkOracle struct {
+	cfg       replicaoracle.Config
+	locFilter roachpb.Locality
+}
+
+var _ replicaoracle.Oracle = bulkOracle{}
+
+// NewBulkOracle returns an oracle for planning bulk operations, which will plan
+// balancing random across all replicas if follower reads are enabled.
+func NewBulkOracle(cfg replicaoracle.Config, locFilter roachpb.Locality) replicaoracle.Oracle {
+	return bulkOracle{cfg: cfg, locFilter: locFilter}
+}
+
+// ChoosePreferredReplica implements the replicaoracle.Oracle interface.
+func (r bulkOracle) ChoosePreferredReplica(
```

Comment:
small test for this function?

Reply:
Added the smallest of tests -- verifies that it doesn't return an error and when a loc filter is used it picks the right node. Doing anything else feels like it'd just be testing the rng though.